services/data-info/src/data_info/services/sharing.clj

```diff
@@ -89,6 +93,52 @@
      :skipped     (map #(dissoc % :skipped) (:skipped share-recs))
      :permission  perm}))
 
+(defn- remove-inherit-bit?
+  [cm user fpath]
+  (empty? (remove (comp (conj (set (cfg/irods-admins)) user) :user)
+                  (list-user-perms cm fpath))))
+
+(defn- unshare-dir
+  "Removes the inherit bit from a directory if the directory is no longer shared with any accounts
+   other than iRODS administrative accounts."
+  [cm user unshare-with fpath]
+  (when (remove-inherit-bit? cm user fpath)
+    (log/warn "Removing inherit bit on" fpath)
+    (remove-inherits cm fpath)))
+
+(defn- unshare-path*
+  "Removes permissions for a user to access a path.  This consists of several steps:
+
+       1. Remove the access permissions for the user.  This is done recursively in case the path
+          being unshared is a directory.
+
+       2. If the item being unshared is a directory, perform any directory-specific unsharing
+          steps that are required.
+
+       3. Remove the user's read permissions for parent directories in which the user no longer has
+          access to any other files or subdirectories."
+  [cm user unshare-with fpath]
+  (let [base-dirs #{(ft/rm-last-slash (paths/user-home-dir user)) (trash-base-dir (:zone cm) user)}]
```

Comment:
It might be worth adding a comment that says why this one is the requesting user's home directory rather than the containing home directory like with sharing (which I guess is just that it's fine to revoke read permissions on someone else's home dir).